manifests/params.pp

```diff
@@ -10,7 +10,7 @@
   $package_ensure    = 'latest'
   $ui_package_name   = 'consul_ui'
   $ui_package_ensure = 'latest'
-  $version = '0.5.0'
+  $version = hiera('consul::version', '0.5.0')  # ensure defaults resolve from hiera as well
```

Comment:
I'm also going to veto an explicit heira lookup here.

I don't understand why `$consul::version` in hiera would not be picked up using the native hiera bindings?

It may be counter-intuitive to you, but this is the documented way that variable precedence takes place:
https://docs.puppetlabs.com/hiera/1/puppet.html#automatic-parameter-lookup